src/Features/Core/Portable/GoToDefinition/AbstractGoToDefinitionSymbolService.cs

```diff
@@ -31,6 +33,20 @@ public async Task<(ISymbol?, TextSpan)> GetSymbolAndBoundSpanAsync(Document docu
             return (FindRelatedExplicitlyDeclaredSymbol(symbol, semanticModel.Compilation), semanticInfo.Span);
         }
 
+        public async Task<SyntaxToken?> GetTargetIfControlFlowAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            Debug.Assert(syntaxTree is not null);
+
+            var language = syntaxTree.GetRoot(cancellationToken).Language;
+            var syntaxFacts = document.Project.Solution.Workspace.Services.GetLanguageServices(language).GetService<ISyntaxFactsService>();
+            Debug.Assert(syntaxFacts is not null);
+
+            var token = await syntaxTree.GetTouchingTokenAsync(position, syntaxFacts.IsBindableToken, cancellationToken, findInsideTrivia: true).ConfigureAwait(false);
```

Comment:
this is good.  please make a test where we have `continue$$;` to validate that we work properly here.